bundles/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/handler/SerialDeviceHandler.java

```diff
@@ -0,0 +1,179 @@
+
+    private final ValueTransformationProvider valueTransformationProvider;
+
+    private @Nullable Pattern devicePattern;
+
+    private @Nullable String lastValue;
+
+    private final Map<ChannelUID, DeviceChannel> channels = new HashMap<>();
+
+    public SerialDeviceHandler(final Thing thing, final ValueTransformationProvider valueTransformationProvider) {
+        super(thing);
+        this.valueTransformationProvider = valueTransformationProvider;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (command instanceof RefreshType) {
+            final String lastValue = this.lastValue;
+
+            if (lastValue != null) {
+                final DeviceChannel channel = channels.get(channelUID);
+                if (channel != null) {
+                    refresh(channelUID, channel, lastValue);
+                }
+            }
+        } else {
+            final DeviceChannel channel = channels.get(channelUID);
+            if (channel != null) {
+                final Bridge bridge = getBridge();
+                if (bridge != null) {
+                    final SerialBridgeHandler handler = (SerialBridgeHandler) bridge.getHandler();
+                    if (handler != null) {
+                        channel.mapCommand(command).ifPresent(value -> handler.writeString(value));
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final SerialDeviceConfiguration config = getConfigAs(SerialDeviceConfiguration.class);
+
+        try {
+            devicePattern = Pattern.compile(config.patternMatch);
+        } catch (final PatternSyntaxException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Invalid device pattern: " + e.getMessage());
+            return;
+        }
+
+        for (final Channel c : getThing().getChannels()) {
+            final ChannelTypeUID type = c.getChannelTypeUID();
+            if (type != null) {
+                final ChannelConfig channelConfig = c.getConfiguration().as(ChannelConfig.class);
+                try {
+                    final DeviceChannel deviceChannel = DeviceChannelFactory
+                            .createDeviceChannel(valueTransformationProvider, channelConfig, type.getId());
+                    if (deviceChannel != null) {
+                        channels.put(c.getUID(), deviceChannel);
+                    }
+                } catch (final IllegalArgumentException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Configuration error for channel " + c.getUID().getId() + ": " + e.getMessage());
+                    return;
+                }
+            }
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+        bridgeStatusChanged(getBridgeStatus());
```

Comment:
Why is the bridge status updated when the thing has been initialized?

Reply:
In fact this sets the thing status depending on the bridge status by calling `bridgeStatusChanged` with the current bridge status. The other option would be to duplicate the code from `bridgeStatusChanged` here.